src/bin/coreutils.rs

```diff
@@ -50,6 +51,52 @@ fn name(binary_path: &Path) -> Option<&str> {
     binary_path.file_stem()?.to_str()
 }
 
+fn get_canonical_util_name(util_name: &str) -> &str {
+    match util_name {
+        // uu_test aliases - '[' is an alias for test
+        "[" => "test",
+
+        // hashsum aliases - all these hash commands are aliases for hashsum
+        "md5sum" | "sha1sum" | "sha224sum" | "sha256sum" | "sha384sum" | "sha512sum"
+        | "sha3sum" | "sha3-224sum" | "sha3-256sum" | "sha3-384sum" | "sha3-512sum"
+        | "shake128sum" | "shake256sum" | "b2sum" | "b3sum" => "hashsum",
+
+        "dir" => "ls", // dir is an alias for ls
+
+        // Default case - return the util name as is
+        _ => util_name,
+    }
+}
+
+fn find_prefixed_util<'a>(
+    binary_name: &str,
+    mut util_keys: impl Iterator<Item = &'a str>,
+) -> Option<&'a str> {
+    util_keys.find(|util| {
+        binary_name.ends_with(*util)
+            && binary_name.len() > util.len() // Ensure there's actually a prefix
+            && !binary_name[..binary_name.len() - (*util).len()]
+                .ends_with(char::is_alphanumeric)
+    })
+}
+
+fn setup_localization_or_exit(util_name: &str) {
+    match locale::setup_localization(get_canonical_util_name(util_name)) {
+        Ok(()) => {}
+        Err(uucore::locale::LocalizationError::ParseResource {
+            error: err_msg,
+            snippet,
+        }) => {
+            eprintln!("Localization parse error at {snippet}: {err_msg}");
+            process::exit(99);
+        }
+        Err(other) => {
+            eprintln!("Could not init the localization system: {other}");
+            process::exit(99);
+        }
+    }
```

Comment:
I think you can simplify it with something like:
```suggestion
    locale::setup_localization(get_canonical_util_name(util_name)).unwrap_or_else(|err| {
        match err {
            uucore::locale::LocalizationError::ParseResource {
                error: err_msg,
                snippet,
            } => eprintln!("Localization parse error at {snippet}: {err_msg}"),
            other => eprintln!("Could not init the localization system: {other}"),
        }
        process::exit(99)
    });
```